core/src/main/java/hudson/ExtensionFinder.java

```diff
@@ -666,7 +668,7 @@ public synchronized ExtensionComponentSet refresh() {
 
             List<IndexItem<Extension,Object>> r = new ArrayList<>(old);
             r.addAll(delta);
-            indices = ImmutableList.copyOf(r);
+            indices = Collections.unmodifiableList(r);
```

Comment:
`r` was declared within this method and goes out of scope at the end of the method, so nothing else can modify it. We do not need to copy it.